### PR TITLE
PUBAPI-1408 ListVolumes endpoint should validate the "predicate" parameter

### PR DESCRIPTION
**Pull request by misterdjules**

### CODE REVIEW

**Comment by misterdjules** [comment]
This PR depends on VOLAPI changes submitted at https://cr.joyent.us/#/c/2101.

File: test/volumes-list-predicate-validation.test.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+ */
+
+var libuuid = require('libuuid');
+var test = require('tape').test;
+var vasync = require('vasync');
+
+var common = require('./common');
+var mod_config = require('../lib/config.js');
+var CONFIG = mod_config.configure();
+if (CONFIG.experimental_nfs_shared_volumes !== true) {
```
Comment (joshwilsdon) [inline test/volumes-list-predicate-validation.test.js]: Don't we ordinarily have a blank line after the `var`s? Just an observation.

Comment (misterdjules) [inline test/volumes-list-predicate-validation.test.js]: Yes, will fix, thanks!

Comment (misterdjules) [inline test/volumes-list-predicate-validation.test.js]: Updated this PR with a new commit.

Comment (joshwilsdon) [inline test/volumes-list-predicate-validation.test.js]: The previous change was good. This is better! 🎆

**Comment by misterdjules** [comment]
Merged with https://github.com/joyent/sdc-cloudapi/commit/f0f585114eeeb063ec17674f8c7db0fa3d502b67.

**Pull request closed without merging**